src/Dialog/Dialog.js

```diff
@@ -1,56 +1,58 @@
-// @flow weak
+// @flow
 
 import React, { Element } from 'react';
 import classNames from 'classnames';
 import { createStyleSheet } from 'jss-theme-reactor';
-import customPropTypes from '../utils/customPropTypes';
+import withStyles from '../styles/withStyles';
 import Modal from '../internal/Modal';
 import Fade from '../transitions/Fade';
 import { duration } from '../styles/transitions';
 import Paper from '../Paper';
 
-export const styleSheet = createStyleSheet('MuiDialog', (theme) => {
-  return {
-    modal: {
-      justifyContent: 'center',
-      alignItems: 'center',
+export const styleSheet = createStyleSheet('MuiDialog', (theme) => ({
+  modal: {
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  dialog: {
+    display: 'flex',
+    flexDirection: 'column',
+    flex: '0 1 auto',
+    position: 'relative',
+    width: '75%',
+    maxHeight: '90vh',
+    '&:focus': {
+      outline: 'none',
     },
-    dialog: {
-      display: 'flex',
-      flexDirection: 'column',
-      flex: '0 1 auto',
-      position: 'relative',
-      width: '75%',
-      maxHeight: '90vh',
-      '&:focus': {
-        outline: 'none',
-      },
-    },
-    'dialogWidth-xs': {
-      maxWidth: theme.breakpoints.getWidth('xs'),
-    },
-    'dialogWidth-sm': {
-      maxWidth: theme.breakpoints.getWidth('sm'),
-    },
-    'dialogWidth-md': {
-      maxWidth: theme.breakpoints.getWidth('md'),
-    },
-    fullScreen: {
-      width: '100%',
-      maxWidth: '100%',
-      height: '100%',
-      maxHeight: '100%',
-      borderRadius: 0,
-    },
-  };
-});
+  },
+  'dialogWidth-xs': {
+    maxWidth: theme.breakpoints.getWidth('xs'),
+  },
+  'dialogWidth-sm': {
+    maxWidth: theme.breakpoints.getWidth('sm'),
+  },
+  'dialogWidth-md': {
+    maxWidth: theme.breakpoints.getWidth('md'),
+  },
+  fullScreen: {
+    width: '100%',
+    maxWidth: '100%',
+    height: '100%',
+    maxHeight: '100%',
+    borderRadius: 0,
+  },
+}));
 
 type Props = {
   /**
    * Dialog children, usually the included sub-components.
    */
   children?: Element<*>,
   /**
+   * Useful to extend the style applied to components.
+   */
+  classes: any,
```

Comment:
This should probably be `classes: Object`, because it won't be a Function or scalar type.